tests/run/posonly.pyx

```diff
@@ -0,0 +1,323 @@
+    TypeError: test_positional_only_and_optional_arg_invalid_calls() takes at most 3 positional arguments (4 given)
+    """
+    pass
+
+def test_positional_only_invalid_calls(a, b, /):
+    """
+    >>> test_positional_only_invalid_calls(1, 2)
+    >>> test_positional_only_invalid_calls(1)
+    Traceback (most recent call last):
+    TypeError: test_positional_only_invalid_calls() takes exactly 2 positional arguments (1 given)
+    >>> test_positional_only_invalid_calls()
+    Traceback (most recent call last):
+    TypeError: test_positional_only_invalid_calls() takes exactly 2 positional arguments (0 given)
+    >>> test_positional_only_invalid_calls(1, 2, 3)
+    Traceback (most recent call last):
+    TypeError: test_positional_only_invalid_calls() takes exactly 2 positional arguments (3 given)
+    """
+    pass
+
+def test_positional_only_with_optional_invalid_calls(a, b=2, /):
+    """
+    >>> test_positional_only_with_optional_invalid_calls(1)
+    >>> test_positional_only_with_optional_invalid_calls()
+    Traceback (most recent call last):
+    TypeError: test_positional_only_with_optional_invalid_calls() takes at least 1 positional argument (0 given)
+    >>> test_positional_only_with_optional_invalid_calls(1, 2, 3)
+    Traceback (most recent call last):
+    TypeError: test_positional_only_with_optional_invalid_calls() takes at most 2 positional arguments (3 given)
+    """
+    pass
+
+def test_no_standard_args_usage(a, b, /, *, c):
+    """
+    >>> test_no_standard_args_usage(1, 2, c=3)
+    >>> test_no_standard_args_usage(1, b=2, c=3)
+    Traceback (most recent call last):
+    TypeError: test_no_standard_args_usage() takes exactly 2 positional arguments (1 given)
+    """
+    pass
+
+#def test_change_default_pos_only():
+# TODO: probably remove this, since we have no __defaults__ in Cython?
+#    """
+#    >>> test_change_default_pos_only()
+#    True
+#    True
+#    """
+#    def f(a, b=2, /, c=3):
+#        return a + b + c
+#
+#    print((2,3) == f.__defaults__)
+#    f.__defaults__ = (1, 2, 3)
+#    print(f(1, 2, 3) == 6)
+
+def test_lambdas():
+    """
+    >>> test_lambdas()
+    3
+    3
+    3
+    3
+    3
+    """
+    x = lambda a, /, b: a + b
+    print(x(1,2))
+    print(x(1,b=2))
+
+    x = lambda a, /, b=2: a + b
+    print(x(1))
+
+    x = lambda a, b, /: a + b
+    print(x(1, 2))
+
+    x = lambda a, b, /, : a + b
+    print(x(1, 2))
+
+
+#TODO: need to implement this in the 'error' test
+#def test_invalid_syntax_lambda(self):
+#    lambda a, b = 5, /, c: None
+#    lambda a = 5, b, /, c: None
+#    lambda a = 5, b, /: None
+#    lambda a, /, a: None
+#    lambda a, /, *, a: None
+#    lambda *args, /: None
+#    lambda *args, a, /: None
+#    lambda **kwargs, /: None
+#    lambda /, a = 1: None
+#    lambda /, a: None
+#    lambda /: None
+#    lambda *, a, /: None
+#    lambda *, /, a: None
+
+class Example:
+    def f(self, a, b, /):
+        return a, b
+
+def test_posonly_methods():
```

Comment:
You can also test this in the docstring of the class, no need for a separate test function here.

Reply:
Done